infrastructure/storage/src/lmdb_store/store.rs

```diff
@@ -359,11 +368,61 @@ impl LMDBStore {
     pub fn env(&self) -> Arc<Environment> {
         self.env.clone()
     }
+
+    /// Returns the initial size in bytes of the LMDBStore. This is the same size used when increasing the size of the
+    /// LMDB Environment mapsize.
+    pub fn get_init_size(&self) -> usize {
+        self.env_init_size_bytes
+    }
+
+    /// Resizes the provided LMDB Environment mapsize to the new size given in bytes.
+    pub fn resize_env(env: &Environment, new_size_bytes: usize) -> Result<(), LMDBError> {
+        let previous_size = env.info()?.mapsize;
+
+        unsafe {
+            env.set_mapsize(new_size_bytes)?;
+        }
+
+        trace!(
+            target: LOG_TARGET,
+            "Resized LMDB env from {} to {}. Difference: {}",
+            previous_size,
+            new_size_bytes,
+            new_size_bytes - previous_size
+        );
+
+        Ok(())
+    }
+
+    /// Returns a result tuple indicating if the LMDB Environment should be resized, along with the new size in bytes.
+    /// Currently uses a constant resize threshold.
+    pub fn resize_required(env: &Environment, init_size_bytes: usize) -> Result<(bool, usize), LMDBError> {
+        let env_info = env.info()?;
+        let psize = env.stat()?.psize as usize;
+        let size_used_bytes = psize * env_info.last_pgno;
+        let new_size = size_used_bytes + init_size_bytes;
+        let needs_resize = should_resize(size_used_bytes, env_info.mapsize, RESIZE_THRESHOLD);
+
+        trace!(
+            target: LOG_TARGET,
+            "LMDB mapsize: {}, bytes used: {}, bytes remaining: {}, used: {:.*}, threshold: {}, resize required: {}.",
+            env_info.mapsize,
+            size_used_bytes,
+            env_info.mapsize - size_used_bytes,
+            4, // decimal places for "used"
+            size_used_bytes as f64 / env_info.mapsize as f64,
+            RESIZE_THRESHOLD,
+            needs_resize.to_string()
+        );
```

Comment:
```suggestion
        let needs_resize = should_resize(size_used_bytes, env_info.mapsize, RESIZE_THRESHOLD);

        let new_size = match needs_resize {
            true => {
                debug!(
                    target: LOG_TARGET,
                    "LMDB mapsize resize required: {}, bytes used: {}, bytes remaining: {}, used: {:.*}, threshold: \
                     {}.",
                    env_info.mapsize,
                    size_used_bytes,
                    env_info.mapsize - size_used_bytes,
                    4, // decimal places for "used"
                    size_used_bytes as f64 / env_info.mapsize as f64,
                    RESIZE_THRESHOLD,
                );
                size_used_bytes + init_size_bytes
            },
            false => init_size_bytes,
        };
```

Reply:
Thanks for the suggestion Hansie - I'll certainly take a look at this. It shouldn't return `init_size_bytes` though - as the env size may have already grown beyond that and so this would actually try to decrease the mapsize.